webhook/webhook_integration_test.go

```diff
@@ -105,8 +105,69 @@ func TestMissingContentType(t *testing.T) {
 	metricstest.CheckStatsNotReported(t, requestCountName, requestLatenciesName)
 }
 
+func TestMissingContentTypeCustomSecret(t *testing.T) {
+	defaultOptions := newCustomOptions()
+	certresources.MakeSecret = customSecretWithOverrides
```

Comment:
This might cause flakiness when running tests in parallel.

We should just have this test perform it's own setup (ie. create it's own cert etc)

Also the test name and assertions is a bit misleading `TestMissingContentTypeCustomSecret` - we really want to assert that the right certificate is presented. Thus we should skip stuff that's not relevant

Reply:
I copied the logic in [certificates_test.go](certresources.MakeSecret = certresources.MakeSecretInternal) so probably that is unsafe too?

Reply:
Anyway I updated the test to be more compact and have its own setup.